### PR TITLE
Fixed more-button wording when there are <15 posts

### LINKED ISSUE #6093: Incorrect wording / functionality of the More button in Mobile view
The introduction of v0.5.0.0 seems to have brought a new issue in with regards to the wording and/or functionality of the More button in Mobile view.

Steps to reproduce:

1. Switch to mobile view
2. Head over to something like */tags/exmachina* and you’ll see a small number of posts
3. At the bottom, where the 'More' button usually appears, you'll see a button saying, “There are no posts yet.”

Well, there *are* posts there...

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/21655766-incorrect-wording-functionality-of-the-more-button-in-mobile-view?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Faldrian**
Added the scenario where there are <15 posts and the user should see "You have reached the end of the stream.".

"More" is only shown when there are 15 (maximum posts on a page) or more (we won't know, since the amount of items is limited by 15) posts. This can lead to one page showing 15 posts and the "More"-Button and when the button is clicked the next page presents a "You have reached the end of the stream."-Message.
Technically correct and without additional data from the database / an extra query I don't think this could get more convenient.

This fixes #6093

### CODE REVIEW

File: features/mobile/more-button.feature [hunk 1]
```diff
@@ -0,0 +1,71 @@
+@javascript @mobile
+Feature: posting from the mobile main page
```
Comment (svbergerem) [inline features/mobile/more-button.feature]: I think you forgot to change this line. ;-)

File: features/mobile/more-button.feature [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      And "bob@bob.bob" has a public post with text "post 6"
+      And "bob@bob.bob" has a public post with text "post 7"
+      And "bob@bob.bob" has a public post with text "post 8"
+      And "bob@bob.bob" has a public post with text "post 9"
+      And "bob@bob.bob" has a public post with text "post 10"
+      And "bob@bob.bob" has a public post with text "post 11"
+      And "bob@bob.bob" has a public post with text "post 12"
+      And "bob@bob.bob" has a public post with text "post 13"
+      And "bob@bob.bob" has a public post with text "post 14"
+      And "bob@bob.bob" has a public post with text "post 15"
```
Comment (svbergerem) [inline features/mobile/more-button.feature]: I think this should be a new step definition. You can find a similar one here https://github.com/Faldrian/diaspora/blob/6093-fix-mobile-pagination-wording/features/step_definitions/user_steps.rb#L102.

This one could be
```ruby
Given /^(\d) public posts from "([^\"]*)"$/ do |n_posts, username|
```

Comment (Faldrian) [inline features/mobile/more-button.feature]: Okay, will do. In the past I had added a step and got the comment that there is already a step doing what I wanted, so this time I wanted to do as much as possible with existing steps.

File: features/step_definitions/posts_steps.rb [hunk 3]
```diff
@@ -27,6 +27,13 @@
   user.post(:status_message, :text => text, :public => true, :to => user.aspect_ids)
 end
 
+Given /^there are (\d+) public posts from "([^"]*)"$/ do |n_posts, email|
+  user = User.find_by_email(email)
+  (1..n_posts.to_i).each do |n|
+    user.post(:status_message, :text => "post nr. #{n}", :public => true, :to => user.aspect_ids)
```
Comment (svbergerem) [inline features/step_definitions/posts_steps.rb]: I don't know why hound isn't complaining here but I think that you should use the new ruby hash syntax instead.

Comment (Faldrian) [inline features/step_definitions/posts_steps.rb]: what syntax? I looked for how to evaluate a variable inside a string and found this. How to do it "right"?

Comment (jhass) [inline features/step_definitions/posts_steps.rb]: Steffen is referring to the shorter syntax for hashes with Symbol keys introduced with Ruby 1.9: `user.post(:status_message, text: "post nr. #{n}", public: true, to: user.aspect_ids)`

**Comment by Faldrian** [comment]
travis is stuck somewhere again

**Comment by svbergerem** [comment]
Thank you. :)

**Comment by RichKavanagh** [comment]
Great work @Faldrian - thanks man! :)

**Pull request closed without merging**